### DISCUSSION TITLE
Can Adguard listen to Unix sockets?

### DISCUSSION THREAD

**Discussion by DavidOsipov** [AdguardTeam/AdGuardHome#3810] [Q&A] [answered]
Hi, guys! I'm curious, whether the Adguard Home is able to listen to queries from unix sockets? And if Adguard Home is installed as a snap, how can I make it listen to a Unix socket? 

Why am I so curious about it - It seems, that unix sockets can [be much faster](https://newbedev.com/tcp-loopback-connection-vs-unix-domain-socket-performance) (~50% faster) than tcp standard loopback

**Comment by ainar-g** [comment, marked as answer]
AGH depends on our `dnsproxy` module, which in turn only supports TCP and UDP connections.  If someone really wanted, I think they could add support for Unix domain sockets, but that would take a lot of time and effort, and the module will probably be heavily rewritten or replaced in the future any way.

**Reply by DavidOsipov** [reply]
Thank you for your answer! Would it be a good idea to create an Issue, so that this feature can be stored in a backlog at least?

**Reply by ainar-g** [reply]
Feel free to create an issue, yes.
